question-servers/freeform.js

```diff
@@ -1468,12 +1461,21 @@ module.exports = {
         ];
         await chunks.ensureChunksForCourseAsync(course.id, chunksToLoad);
 
+        // The `*Host` values here refer to the paths relative to PrairieLearn;
+        // the other values refer to the paths as they will be seen by the worker
+        // that actually executes the question.
+        const courseDirectory = config.workersExecutionMode === 'internal' ? coursePath : '/course';
+        const courseDirectoryHost = coursePath;
+        const questionDirectory = path.join(courseDirectory, 'questions', question.directory);
+        const questionDirectoryHost = path.join(coursePath, 'questions', question.directory);
```

Comment:
We need to differentiate between paths as PrairieLearn sees them and files as the worker that executes code sees them. For `workersExecutionrMode === 'internal'`, these paths are the same, but they differ if questions are executing in a Docker container.

Throughout this file, we construct and log files paths for errors (for instance, if `server.py` misbehaves). In those cases, I opted to use the non-host version. This means logs will include things like `/course/questions/foo/server.py` instead of `/data1/courses/pl-cs225/questions/foo/server.py`. This seems better as it doesn't leak our internal directory structure. Again, wanted to call this out in case of differing opinions.

Reply:
I'm very much in favor of this! Only showing "inside executor container" paths to the user will be great.